app/code/Magento/NewsletterApi/Exception/AlreadySubscribedException.php

```diff
@@ -0,0 +1,14 @@
+<?php
+/**
```

Comment:
Well, you're suggesting us to use M2's exceptions, when you guys have created your own? Not fair.

How would you differentiate between some `LocalizedException` down the stack and a `LocalizedException` that's only triggered if there already is a subscriber? Existing exceptions are too generic than this unique usage.

> Do create and throw custom exceptions if you have an error condition that can be programmatically handled in a different way than any other existing exceptions.

**Every** exception can be handled that way, but that's not the point of specialised exceptions. It's about granular handling. There are cases when you can handle an `AlreadySubscribedException` exception, and choose not to handle a `LocalizedException`, so you throw it further up the stack.

Specific exceptions are a good thing. Don't make us catch and throw generic exceptions. Maybe @davidverholen can extend `\Magento\Framework\Exception\AlreadyExistsException` instead, but I still consider this exception useful. Heck, I'll go as far as saying that every _module_ needs its own exception namespace.